src/test/java/io/vantiq/client/intg/VantiqIntegrationTest.java

```diff
@@ -431,11 +431,15 @@ public void testSubscribeReliableTopic() throws Exception {
         vantiq.subscribe(Vantiq.SystemResources.TOPICS.value(), "/test/topic", null, callback, params);
         callback.waitForCompletion();
         LinkedTreeMap msg = (LinkedTreeMap) callback.getMessage().getBody();
-        String ackId = msg.get("subscriptionName").toString();
-        String requestId = msg.get("requestId").toString();
-        callback.reset();
-        callback.waitForCompletion();
-        assertThat("Connected", callback.isConnected(), is(true));
+        assertThat(msg.get("subscriptionName"), instanceOf(String.class));
+        assertThat(callback.getMessage().getHeaders().get("X-Request-Id"), instanceOf(String.class));
+    
+        // Sometimes the connection occurs during the validations above. Check for completion only if it's not connected
+        if (!callback.isConnected()) {
+            callback.reset();
+            callback.waitForCompletion();
+            assertThat("Connected", callback.isConnected(), is(true));
+        }
```

Comment:
If the connection message comes fast enough, it will occur before the reset. The assert here will then fail since `reset()` clears all the flags used by the test callback class.

Reply:
Might put the assert outside of the IF, just so it is always checked.  Makes test clearer to me.
```suggestion
        // Sometimes the connection occurs during the validations above. 
        // So wait for completion only if it's not already connected
        if (!callback.isConnected()) {
            callback.reset();
            callback.waitForCompletion();
        }
        assertThat("Connected", callback.isConnected(), is(true));
```